File: retos-ts/12_cuanto_cuesta.ts
// Resultado 5 ⭐
function calculatePrice(ornaments: string): number | undefined {
    const cost: { [key:string]: number } = { '*': 1, 'o': 5, '^': 10, '#': 50, '@': 100 }
    let result: number = 0
    let i: number = 1

    for(const o of ornaments){
        result += cost[o] < cost[ornaments[i]]
            ? -cost[o]
            : cost[o]

        i++
    }
    
    return result || undefined
}

File: retos-python/12_cuanto_cuesta.py
# Resultado 
# Espera un undefined pero se puede regresar null
def calculate_price(ornaments: str):
    cost = { '*': 1, 'o': 5, '^': 10, '#': 50, '@': 100 }
    result = 0
    i = 1

    for o in ornaments:
        if i < len(ornaments) and cost[o] < cost[ornaments[i]]:
            result -= cost[o]
        else:
            result += cost[o]
        i += 1

    return result or None

